src/main/java/fi/aalto/cs/intellij/common/Course.java

```diff
@@ -0,0 +1,219 @@
+package fi.aalto.cs.intellij.common;
+
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.jetbrains.annotations.NotNull;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+
+public class Course {
+  @NotNull
+  private final String name;
+
+  @NotNull
+  private final List<Module> modules;
+
+  // Maps ids of required plugins to the names of required plugins.
+  @NotNull
+  private final Map<String, String> requiredPlugins;
```

Comment:
Plugins could also be represented as objects but IMHO, this is totally fine for now.

Reply:
Will keep this in mind if we ever decide to add more specific plugin information (more than just names and ids) to the course configuration file :+1: